internal/http.go

```diff
@@ -0,0 +1,206 @@
+package internal
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	postgresURL    = "host=postgresql.test-project.svc  port=5432 user=curator password=M5rBgWkN8LfjeyI8 dbname=curatordb sslmode=disable"
+	httpServerPort = ":5000"
+)
+
+var (
+	reportLog = ctrl.Log.WithName("report")
+)
+
+type Response struct {
+	Message string `json:"message"`
+}
+
+func InitializeRouter() {
```

Comment:
Reading through this function some more: I think we need to refactor this logic. I think what we'll end up needing is to split this logic into a constructor that's responsible for populating a custom struct for configuring our internal HTTP server, and a method that runs that HTTP server and handles the interrupt signals we'll be watching for.